### ISSUE TITLE
Blurry panel icons sized from 25px to 31px

### ISSUE THREAD

**Issue by firepainting** [linuxmint/cinnamon#9781]
* Cinnamon version: 4.6.7 (Not using the daily builds PPA)
 * Distribution - Mint 20.0

 * Graphics hardware *and* driver used
```
  Device-1: Intel Xeon E3-1200 v3/4th Gen Core Processor Integrated Graphics 
  vendor: Micro-Star MSI driver: i915 v: kernel bus ID: 00:02.0 
  Display: x11 server: X.Org 1.20.8 driver: modesetting unloaded: fbdev,vesa 
  resolution: 1366x768~60Hz 
  OpenGL: renderer: Mesa DRI Intel HD Graphics 4600 (HSW GT2) 
  v: 4.5 Mesa 20.0.8 direct render: Yes 
```

 * 64 bit

 * /home/username/.xsession-errors, /var/log/syslog

Edit 2: Removed attachments for /home/username/.xsession-errors and /var/log/syslog for privacy reasons. Not relevant, anyway.

**Issue**

Icons sized between (but not) 24 and 32px are blurry, i.e, 25, 26, 27, 28, 29, 30, 31px. (24 and 32px icons are crisp.)

**Steps to reproduce**

Open:
`xed admin:///usr/share/cinnamon/cinnamon-settings/modules/cs_panel.py`

Add 25, 26, 27, 28, 29, 30, 31px on the part below:
```
        fullcolor_options = [
            [-1, _("Scale to panel size exactly")],
            [0, _("Scale to panel size optimally")],
            [16, '16px'],
            [22, '22px'],
            [24, '24px'],
            [32, '32px'],
            [48, '48px']
        ]
```
Go to Panel Settings > Panel Appearance > Colored Icon Size > Set to 26px

**Expected behaviour**

Crisp panel icons.

**Other information**

Screenshot of icons in 26px (blurry)

![26px](https://user-images.githubusercontent.com/50393421/102451298-2c162e80-4073-11eb-8298-e5884e35b346.png)

Edit 1: Desktop theme is Mint-Y. 

**Configuration info, if applicable**

N/A

Thank you.

**Comment by NikoKrause** [comment]
That's correct. Therefore the option: Scale to panel size optimally.

The icons are png files and scaling them results in blurry icons. Creating a png file for every possible icon size is a monumental task and won't happen. There are certain icons sizes as 16/24/32 etc. , which are used by most icon themes. And I think there is a reason why svg icons are not used, think it slows down the filemanager or something like that. Can't remember.

**Comment by firepainting** [comment]
Thank you, @NikoKrause.

I didn't know there's a PNG file for each default icon size (16, 22, 24, 32, 48px) and that anything in between would only have to be scaled, and therefore blurring it.

Is there another place I could comment on this (if at all)? I think it's a reasonable point especially for screens with not-so-high resolutions like mine.

Re "scale to panel size optimally," I prefer a bit thicker panel and doing so would render huge icons. 32px is already too big to me.

**Comment by NikoKrause** [comment]
I don't want to sound rude, I'm just trying to be honest on this point.

You can comment on the mint-y-icons repository. It won't change anything though in the near or not so near future. See: https://github.com/linuxmint/mint-y-icons/issues/168

It's a known issue, but creating icons for standard icon sizes to make them look crisp is already a very time consuming task. Creating them for non-standard icon sizes as 25/26/27/28/29/30/31 etc. is a task that nobody would want to tackle nor would have the time for. 

Your current options right now are:
1. You're saying you like a big panel, but 32 px icons are too big for you. Use the 24 px option. But I guess those are too small for you.
2. Change to an icon theme, which provides non-standard icon sizes. I am not aware of any, which does. Or change to an icon theme, which uses svg files instead of png files. 
3. Create your own version of Mint-Y icons with svgs. You find the src files in the mint-y-icons repository.

**Comment by firepainting** [comment]
Thanks again!

No, you don't sound rude at all but I apologize if I did sound rude... I only surmised that there may be a better place to talk about this issue, somewhere "upstream" if that makes sense. I appreciate the link you referred me to.

In any case I'm not expecting anything to change soon. I now understand how things stand as far as icon sizes are concerned.

It's something but I don't really mind!
